packages/acceptance-tests/pkg-tests-specs/sources/commands/pack.test.js

```diff
@@ -287,5 +288,66 @@ describe(`Commands`, () => {
         await expect(stdout).toMatch(/package\.json/);
       }),
     );
+
+    test(
+      `the filename is non-descriptive by default`,
+      makeTemporaryEnv({
+        name: '@berry/core',
+        version: '0.0.1',
+      }, async ({path, run, source}) => {
+        await run(`install`);
+
+        const {stdout} = await run(`pack`);
+        await expect(stdout).toMatch(/package\.json/);
+        await expect(stdout).toMatch(/Package archive generated in .+?\/package\.tgz/);
+      }),
+    );
+
+    test(
+      `can generate an archive with a descriptive name`,
+      makeTemporaryEnv({
+        name: '@berry/core',
+        version: '0.0.1',
+      }, async ({path, run, source}) => {
+        await run(`install`);
+
+        const {stdout} = await run(`pack`, `--out`, `./%n.tgz`);
+        await expect(stdout).toMatch(/package\.json/);
+        await expect(stdout).toMatch(/Package archive generated in .+?\/berry-core-0\.0\.1\.tgz/);
+      }),
+    );
+
+    test(
+      `the descriptive name doesn't add empty slug parts if the package has no scope`,
+      makeTemporaryEnv({
+        name: 'core',
+        version: '0.0.1',
+      }, async ({path, run, source}) => {
+        await run(`install`);
+
+        const {stdout} = await run(`pack`, `--out`, `./%n.tgz`);
+        await expect(stdout).toMatch(/package\.json/);
+        await expect(stdout).toMatch(/Package archive generated in .+?\/core-0\.0\.1\.tgz/);
+      }),
+    );
+
+    test(
+      `can output the archive in a absolute destination`,
+      makeTemporaryEnv({
+        name: '@berry/core',
+        version: '0.0.1',
+      }, async ({path, run, source}) => {
+        await run(`install`);
+
+        let stderr;
+        try {
+          ({stderr} = await run(`pack`, `--out`, `/artifacts/berry-core.tgz`));
+        } catch (error) {
+          ({stderr} = error);
+        }
+
+        await expect(stderr).toMatch(`no such file or directory, open '${fromPortablePath('/artifacts/berry-core.tgz')}'`);
```

Comment:
Not sure how I'm supposed to tap into the host environment in tests. The code of `fromPortablePath` says it will convert the separators but for some reason it doesn't work in CI.